ai_ref_kits/multimodal_ai_visual_generator/main.py

```diff
@@ -1,17 +1,188 @@
-import app
-import download_and_prepare_models as models
-
-
-def main():
-    models.prepare_llm_model()
-    models.prepare_stable_diffusion_model()
-    models.prepare_voice_activity_detection_model()
-    models.prepare_super_res()
-    models.prepare_whisper()
-    models.prepare_depth_anything_v2()
-
-    app.main()
-
-
-if __name__ == '__main__':
-    main()
\ No newline at end of file
+from fastapi import FastAPI, Request
+from pydantic import BaseModel
+from fastapi.middleware.cors import CORSMiddleware
+from fastapi.responses import JSONResponse
+from pathlib import Path
+from io import BytesIO
+from PIL import Image
+import base64
+import sys
+import yaml
+import subprocess
+import openvino_genai as ov_genai
+
+
+# Import watermark function
+external_utils_path = (Path(__file__).resolve().parents[2] / "demos" / "utils").resolve()
+if not external_utils_path.exists():
+    raise RuntimeError(f"utils folder not found at {external_utils_path}")
+sys.path.append(str(external_utils_path))
+
+from demo_utils import draw_ov_watermark
+
+app = FastAPI()
+
+# Allow CORS (for frontend integration)
+app.add_middleware(
+    CORSMiddleware,
+    allow_origins=["*"],
+    allow_methods=["*"],
+    allow_headers=["*"],
+)
+
+# ---------- Setup Project Paths ----------
+PROJECT_ROOT = Path(__file__).resolve().parent
+CONFIG_PATH = PROJECT_ROOT / "config" / "illustration.yaml"
+
+
+IMAGE_MODEL_TYPE = "flux.1-schnell"
+LLM_MODEL_TYPE = "qwen2-7B"
+PRECISION = "int4"
+
+image_model_dir = PROJECT_ROOT / "models" / f"{IMAGE_MODEL_TYPE}-{PRECISION.upper()}"
+llm_model_dir = PROJECT_ROOT / "models" / f"{LLM_MODEL_TYPE}-{PRECISION.upper()}"
+
+# ---------- Load Config ----------
+with open(CONFIG_PATH, "r") as f:
+    config = yaml.safe_load(f)
+
+# ---------- Lazy load models if available ----------
+image_pipe = None
+llm_pipe = None
+
+if image_model_dir.exists():
+    try:
+        image_pipe = ov_genai.Text2ImagePipeline(image_model_dir, device="GPU")
+        print("Image model loaded.")
+    except Exception as e:
+        print(f"Failed to load Image model: {e}")
+else:
+    print(f"Image model not found at {image_model_dir}")
+
+if llm_model_dir.exists():
+    try:
+        llm_pipe = ov_genai.LLMPipeline(str(llm_model_dir), device="GPU")
+        print("LLM model loaded.")
+    except Exception as e:
+        print(f"Failed to load LLM model: {e}")
+else:
+    print(f"LLM model not found at {llm_model_dir}")
+    
+llm_config = ov_genai.GenerationConfig()
+llm_config.max_new_tokens = 256
+llm_config.apply_chat_template = False
+
+# ---------- Request/Response Schemas ----------
+class PromptRequest(BaseModel):
+    prompt: str
+
+class StoryRequest(BaseModel):
+    prompt: str
+
+# ---------- LLM Endpoint (Story Splitter) ---------
+@app.post("/generate_story_prompts")
+def generate_story_prompts(request: StoryRequest, req: Request):

```

Comment:
This function is very long and complicated. The good idea is to extract some smaller snippets to other functions and call them

Reply:
Updated generate_story_prompts() into smaller helper functions.